base/stats.go

```diff
@@ -0,0 +1,799 @@
+			PulledCount:                 NewIntStat(SubsystemReplication, "sgr_num_docs_pulled", d.dbName, prometheus.CounterValue, 0),
+			PurgedCount:                 NewIntStat(SubsystemReplication, "sgr_num_docs_purged", d.dbName, prometheus.CounterValue, 0),
+			FailedToPullCount:           NewIntStat(SubsystemReplication, "sgr_num_docs_failed_to_pull", d.dbName, prometheus.CounterValue, 0),
+			DeltaReceivedCount:          NewIntStat(SubsystemReplication, "sgr_deltas_recv", d.dbName, prometheus.CounterValue, 0),
+			DeltaRequestedCount:         NewIntStat(SubsystemReplication, "sgr_deltas_requested", d.dbName, prometheus.CounterValue, 0),
+			DocsCheckedReceived:         NewIntStat(SubsystemReplication, "sgr_docs_checked_recv", d.dbName, prometheus.CounterValue, 0),
+			ConflictResolvedLocalCount:  NewIntStat(SubsystemReplication, "sgr_conflict_resolved_local_count", d.dbName, prometheus.CounterValue, 0),
+			ConflictResolvedRemoteCount: NewIntStat(SubsystemReplication, "sgr_conflict_resolved_remote_count", d.dbName, prometheus.CounterValue, 0),
+			ConflictResolvedMergedCount: NewIntStat(SubsystemReplication, "sgr_conflict_resolved_merge_count", d.dbName, prometheus.CounterValue, 0),
+		}
+	}
+
+	return d.DbReplicatorStats[replicationID]
+}
+
+func (d *DbStats) Security() *SecurityStats {
+	return d.SecurityStats
+}
+
+func (d *DbStats) InitSharedBucketImportStats() {
+	if d.SharedBucketImportStats == nil {
+		dbName := d.dbName
+		d.SharedBucketImportStats = &SharedBucketImportStats{
+			ImportCount:          NewIntStat(SubsystemSharedBucketImport, "import_count", dbName, prometheus.CounterValue, 0),
+			ImportCancelCAS:      NewIntStat(SubsystemSharedBucketImport, "import_cancel_cas", dbName, prometheus.CounterValue, 0),
+			ImportErrorCount:     NewIntStat(SubsystemSharedBucketImport, "import_error_count", dbName, prometheus.CounterValue, 0),
+			ImportProcessingTime: NewIntStat(SubsystemSharedBucketImport, "import_processing_time", dbName, prometheus.GaugeValue, 0),
+			ImportHighSeq:        NewIntStat(SubsystemSharedBucketImport, "import_high_seq", dbName, prometheus.CounterValue, 0),
+			ImportPartitions:     NewIntStat(SubsystemSharedBucketImport, "import_partitions", dbName, prometheus.GaugeValue, 0),
+		}
+	}
+}
+
+func (d *DbStats) SharedBucketImport() *SharedBucketImportStats {
+	return d.SharedBucketImportStats
+}
+
+func (d *DbStats) InitQueryStats(useViews bool, queryNames ...string) {
+	d.QueryStats = &QueryStats{
+		Stats: map[string]*QueryStat{},
+	}
+	d.QueryStats.mutex.Lock()
+	for _, queryName := range queryNames {
+		d.initQueryStat(useViews, queryName)
+	}
+	d.QueryStats.mutex.Unlock()
+}
+
+func (d *DbStats) initQueryStat(useViews bool, queryName string) {
+	if _, ok := d.QueryStats.Stats[queryName]; !ok {
+		dbName := d.dbName
+
+		// Prometheus isn't happy with '.'s in the name and the '.'s come from the design doc version. Design doc isn't
+		// reported to prometheus. Only the view name.
```

Comment:
Is this what the current exporter is doing?  It feels like a loss of information if we can't differentiate between view and index stats.  I would have thought that replacing `.` with something like `__` down in newSGWStat would be a preferred approach, but if the current exporter is already doing this we could leave as-is for backward compatibility.

Reply:
Yeah, currently there is no differentiation in the exporter about whether a particular gsi_view stat is for, well, gsi and views. Infact the exporter only currently exports _count stats.